cpp/src/arrow/python/CMakeLists.txt

```diff
@@ -60,7 +60,14 @@ endif()
 
 set(ARROW_PYTHON_SHARED_LINK_LIBS arrow_shared)
 if(WIN32)
-  list(APPEND ARROW_PYTHON_SHARED_LINK_LIBS ${PYTHON_LIBRARIES} ${PYTHON_OTHER_LIBS})
+  if(ARROW_USE_DEPRECATED_FIND_PYTHONLIBS)
+    list(APPEND ARROW_PYTHON_SHARED_LINK_LIBS 
+         $<$<CONFIG:Debug>:${PYTHON_DEBUG_LIBRARIES}> 
+         $<$<CONFIG:Release>:${PYTHON_LIBRARIES}> 
```

Comment:
Can't you factor out the debug/release library switch?

Reply:
Hi, `PYTHON_DEBUG_LIBRARIES` is set on the deprecated `FindPythonLibsNew` script. If I don't use it here, it won't link against `python_d` and the linking stage will fail.